VmAgent.Core/Dependencies/BasicAssetExtractor.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Microsoft.Azure.Gaming.VmAgent.Core.Dependencies.Interfaces;
+using Microsoft.Azure.Gaming.VmAgent.Core.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using VmAgent.Core.Dependencies.Interfaces.Exceptions;
+using VmAgent.Core.Interfaces;
+
+namespace Microsoft.Azure.Gaming.VmAgent.Core.Dependencies
+{
+    public class BasicAssetExtractor : IBasicAssetExtractor
+    {
+        private readonly ISystemOperations _systemOperations;
+        private readonly MultiLogger _logger;
+
+        private readonly string[] LinuxSupportedFileExtensions =
+            { Constants.ZipExtension, Constants.TarExtension, Constants.TarGZipExtension };
+
+        public BasicAssetExtractor(ISystemOperations systemOperations = null, MultiLogger logger = null)
+        {
+            _systemOperations = systemOperations;
+            _logger = logger;
+        }
+
+        public void ExtractAssets(string assetFileName, string targetFolder)
+        {
+            string fileExtension = Path.GetExtension(assetFileName).ToLowerInvariant();
+
+            // If the OS is windows use the native .NET zip extraction (we only support .zip for Windows).
+            if (_systemOperations.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (fileExtension != Constants.ZipExtension)
+                {
+                    throw new AssetExtractionFailedException($"Only Zip format is supported in Windows. Unable to extract {assetFileName}.");
+                }
+
+                if (Directory.Exists(targetFolder))
+                {
+                    Directory.Delete(targetFolder, true);
+                }
+
+                _systemOperations.ExtractToDirectory(assetFileName, targetFolder);
+            }
+            else
```

Comment:
add a comment that we're running as Linux here so it's visible - since we blocked running on MacOS